components/match2/wikis/smite/match_group_input_custom.lua

```diff
@@ -0,0 +1,496 @@
+---
+-- @Liquipedia
+-- wiki=smite
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local DateExt = require('Module:Date/Ext')
+local HeroNames = mw.loadData('Module:GodNames')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Streams = require('Module:Links/Stream')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input')
+local Opponent = Lua.import('Module:Opponent')
+
+local STATUS_SCORE = 'S'
+local STATUS_DRAW = 'D'
+local STATUS_DEFAULT_WIN = 'W'
+local STATUS_FORFEIT = 'FF'
+local STATUS_DISQUALIFIED = 'DQ'
+local STATUS_DEFAULT_LOSS = 'L'
+local _ALLOWED_STATUSES = {
+	STATUS_DRAW,
+	STATUS_DEFAULT_WIN,
+	STATUS_FORFEIT,
+	STATUS_DISQUALIFIED,
+	STATUS_DEFAULT_LOSS,
+}
+local MAX_NUM_OPPONENTS = 2
+local MAX_NUM_PLAYERS = 15
+local MAX_NUM_GAMES = 7
+local DEFAULT_MODE = 'team'
+local NO_SCORE = -99
+local NP_STATUSES = {'skip', 'np', 'canceled', 'cancelled'}
+local DEFAULT_RESULT_TYPE = 'default'
+local NOT_PLAYED_SCORE = -1
+local NO_WINNER = -1
+local SECONDS_UNTIL_FINISHED_EXACT = 30800
+local SECONDS_UNTIL_FINISHED_NOT_EXACT = 86400
+
+local NOW = os.time(os.date('!*t') --[[@as osdateparam]])
+
+-- containers for process helper functions
+local matchFunctions = {}
+local mapFunctions = {}
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+---@param match table
+---@return table
+function CustomMatchGroupInput.processMatch(match)
+	-- process match
+	Table.mergeInto(match, MatchGroupInput.readDate(match.date))
+	match = matchFunctions.getBestOf(match)
+	match = matchFunctions.getScoreFromMapWinners(match)
+	match = matchFunctions.getOpponents(match)
+	match = matchFunctions.getTournamentVars(match)
+	match = matchFunctions.getVodStuff(match)
+
+	-- Adjust map data, especially set participants data
+	match = matchFunctions.adjustMapData(match)
+
+	return match
+end
+
+function matchFunctions.adjustMapData(match)
```

Comment:
Need to remove map names afaik

Reply:
Yes, but it's stored into LPDB

Reply:
> Yes, but it's stored into LPDB otherwise ![image](https://private-user-images.githubusercontent.com/3426850/303319133-d3202f21-72f7-4579-bd6e-a513bd27ac76.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc0NzE5NTQsIm5iZiI6MTcwNzQ3MTY1NCwicGF0aCI6Ii8zNDI2ODUwLzMwMzMxOTEzMy1kMzIwMmYyMS03MmY3LTQ1NzktYmQ2ZS1hNTEzYmQyN2FjNzYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDlUMDk0MDU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjBkMzAxZDZhNzNkOWNjN2U2NTY2MTVmNGNjMzlkMmRhNTJmNTM4NTdkOWFhYjZiMThiZTc3ZGQ3OWRiYzZhNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.HGhFAUNu0wrJdmWSODCmp04bKPvTs3IXd-wuHnYDsMs)

Is there a way **default** can be nil or empty without having the DUMMY_MAP? 

I've tried re-add the Dummy Map and "default" from this section would become nil 
![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/1f29f3bb-7cc0-43b9-9356-7978467d26a2)

Reply:
That looks fine to me